src/ca/mcgill/cs/jetuml/gui/DiagramCanvasController.java

```diff
@@ -564,4 +562,23 @@ private void moveSelection(Point pMousePoint)
 		aLastMousePoint = pMousePoint; 
 		aCanvas.paintPanel();
 	}
-}
\ No newline at end of file
+	
+	/**
+	 * When the shift key is pressed, perform node attachment or detachment if possible
+	 */
+	public void keyPressed()
+	{

```

Comment:
Check if `aCanvas.getDiagram().getType() != DiagramType.CLASS` and return immediately.

Then, downcast `aDiagramBuilder` to `ClassDiagramBuilder`